Tools/Cli-LoRa-Device-Provisioning/NetIdHelper.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Cli_LoRa_Device_Provisioning
+{
+    using System;
+
+    public static class NetIdHelper
+    {
+        /// <summary>
+        /// Set the NetworkId Part of a devAddr with a given network Id.
+        /// </summary>
+        public static string SetNwkIdPart(string currentDevAddr, string networkId, ConfigurationHelper configurationHelper)
+        {
+            var netIdLastByte = configurationHelper.NetId.Substring(configurationHelper.NetId.Length - 2, 2);
```

Comment:
you want to check the NetId before using Substring to ensure it is not null and has at least a length of 2

Reply:
Even though it can never be null? Upon initialization, it is set to the user's setting or defaulted to '000001'...?

Reply:
In my opinion, since this is a public API, better be safe than sorry. :) If caller code changes your implementation should stay stable.

Reply:
I do a validation of configurationHelper.NetId now with my ValidationHelper.ValidateHexStringTwinProperty method first. If all fails, I return the default NetId of 1.